.github/workflows/semgrep-ci.yml

```diff
@@ -24,15 +24,42 @@ env:
   SEMGREP_ARGS: --error --quiet
 
 jobs:
+  semgrep-validate:
```

Comment:
This looks fine to me. I do wonder if we could combine these into steps of a single job, rather than distinct ones, but it's not something I'd hold off on merging for.

Reply:
I considered validating all of the Semgrep files in this job, and then making all the jobs depend on it, but figured this was fewer changes